config/settings.js

```diff
@@ -0,0 +1,81 @@
+// Setttings
+
+// Adapted from: https://github.com/rancher/ui/blob/08c379a9529f740666a704b52522a468986c3520/lib/shared/addon/utils/constants.js#L564
+
+// Setting IDs
+const SETTING = {
+  IMAGE_RANCHER:                    'server-image',
+  VERSION_RANCHER:                  'server-version',
+  VERSION_COMPOSE:                  'compose-version',
+  VERSION_CLI:                      'cli-version',
+  VERSION_MACHINE:                  'machine-version',
+  VERSION_HELM:                     'helm-version',
+  VERSIONS_K8S:                     'k8s-version-to-images',
+  VERSION_RKE_K8S_DEFAULT:          'k8s-version',
+  VERSION_K8S_SUPPORTED_RANGE:      'ui-k8s-supported-versions-range',
+  VERSION_SYSTEM_K8S_DEFAULT_RANGE: 'ui-k8s-default-version-range',
+
+  CLI_URL:                          {
+    DARWIN:                         'cli-url-darwin',
+    WINDOWS:                        'cli-url-windows',
+    LINUX:                          'cli-url-linux',
+  },
+
+  API_HOST:                         'api-host',
+  CA_CERTS:                         'cacerts',
+
+  HIDE_LOCAL_CLUSTER:               'hide-local-cluster',
+  // CLUSTER_DEFAULTS:              'cluster-defaults',
+  AUTH_TOKEN_MAX_TTL_MINUTES:       'auth-token-max-ttl-minutes',
+  ENGINE_URL:                       'engine-install-url',
+  ENGINE_ISO_URL:                   'engine-iso-url',
+  FIRST_LOGIN:                      'first-login',
+  INGRESS_IP_DOMAIN:                'ingress-ip-domain',
+  PL:                               'ui-pl',
+  PL_RANCHER_VALUE:                 'rancher',
+  UI_BANNERS:                       'ui-banners',
+  UI_ISSUES:                        'ui-issues',
```

Comment:
There was some oddness with this. Editing the default blank value results in the ui showing it as a custom value. Assigned some junk value `dsgfgdf` and it fails to show this as `modified` in the list. Returning to edit the value and it's correctly marked as custom.

Reply:
Yeah, fixed.